marshmallow/fields.py

```diff
@@ -274,6 +297,10 @@ def _add_to_schema(self, field_name, schema):
         """
         self.parent = self.parent or schema
         self.name = self.name or field_name
+        # Bind registered method validators so that the `self` argument
+        # is the Schema
+        for validator in self._method_validators:
+            self.validators.append(functools.partial(validator, schema))
```

Comment:
Not that I forsee a huge problem, but won't this bind the validator to the first schema it's on, rather than to the inheritance chain, unless I missed some Juju in SchemaMeta. I can't test my hunch as I'm mobile, but I think this shouldn't produce the expected results...

```
class MySchema(Schema):
    bar = DecimalField()

    @bar.validator
    def error(self, *a, **k):
        raise ValidationError(self.__class__.__name__)


class SubSchema(MySchema):
    pass

SubSchema.dump(dotteddict(bar=4)).errors 
```

But I *could* be wrong. Another sticky part could be child schemas that have stricter validation needs. How would this react to:

```
class SubSchema(MySchema):
    @MySchema.bar.validates
    def stricter a validation(...):
        ...
```